java/arcs/core/data/proto/TypeProtoDecoders.kt

```diff
@@ -30,7 +30,15 @@ fun PrimitiveTypeProto.decode() = when (this) {
 }
 
 /** Converts a [PrimitiveTypeProto] protobuf instance into a Kotlin [FieldType] instance. */
-fun PrimitiveTypeProto.decodeAsFieldType(): FieldType.Primitive = FieldType.Primitive(decode())
+fun PrimitiveTypeProto.decodeAsFieldType() = FieldType.Primitive(decode())
+
+/** Converts a [ReferenceTypeProto] protobuf instance into a Kotlin [FieldType] instance. */
+fun ReferenceTypeProto.decodeAsFieldType() = FieldType.EntityRef(decode().entitySchema!!.hash)
+
+/** Converts a [TupleTypeProto] protobuf instance into a Kotlin [FieldType] instance. */
+fun TupleTypeProto.decodeAsFieldType(): FieldType.Tuple = FieldType.Tuple(
```

Comment:
`...decodeAsFieldType() = FieldType.Tuple(elementsList.map { it.decodeAsFieldType() })`

Reply:
I need to add the return type `FieldType.Tuple` in order to solve an infinite type recursion problem in Kotlin Type Inference. 

When I remove this, I get errors on line 56 (decoding a Tuple as field type).